FitnessClub/Data/BLL/Services/CoachService.cs

```diff
@@ -0,0 +1,135 @@
+﻿using FitnessClub.Data.BLL.Interfaces;
+using FitnessClub.Data.DAL.Interfaces;
+using FitnessClub.Data.DAL.Utility;
+using FitnessClub.Data.Models.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FitnessClub.Data.BLL.Services
+{
+    public class CoachService : EmployeeService, ICoachService
+    {
+        private readonly ISessionRepository sessionRepository;
+        private readonly ICoachRepository coachRepository;
+        private readonly int userId;
+        public CoachService(
+            ISessionRepository sessionRepository,
+            IEmployeeRepository employeeRepository,
+            ICoachRepository coachRepository,
+            UserResolverService userResolverService
+            ) : base(sessionRepository, employeeRepository, userResolverService)
+        {
+            this.sessionRepository = sessionRepository;
+            this.coachRepository = coachRepository;
+            userId = userResolverService.GetUserId();
+        }
+
+        public async Task AssignToSession(int coachId, int sessionId)
+        {
+            var coach = await coachRepository.GetById(coachId);
+            var session = sessionRepository.FindWithCoach(sessionId);
+            if (coach == null )
+            {
+                Serilog.Log.Information($"Couldn't find the coach with id {coachId}.");
+                return;
+            }
+            else if (session == null)
+            {
+                Serilog.Log.Information($"Couldn't find the session with id {sessionId}.");
+                return;
+            }
+            if (session.Coach != null)
+            {
+                Serilog.Log.Information($"Session already has assigned coach. Removed coach with id {session.Coach.PersonID} from the session and assigned the new one.");
+            }
+            session.Coach = coach;
+            session.CoachID = coach.PersonID;
```

Comment:
Why does SessionModel need CoachID since it already has Coach?

Reply:
This has already been done differently (commited to dev branch), now the view model is a proxy over the session, keeping private readonly instance of the session, and there are properties that access the private session's members.